File: tempo-rise-webapp/src/app/core/models/promotion.ts
import { Product } from "./product";

export type PromotionType = "FIXED" | "PERCENTAGE";
export interface Promotion {
    id: number;
    nom: string;
    description: string;
    reduction: number;
    type: PromotionType;
    dateDebut: string;
    dateFin: string;
    status:string,
    produitId:number,
    produit?:Product
}

File: tempo-rise-api/src/main/java/tn/temporise/domain/model/Promotion.java
package tn.temporise.domain.model;



import lombok.Builder;

import java.time.LocalDateTime;

@Builder(toBuilder = true)

public record Promotion(
        Long id,
        String nom,
        String description,
        double reduction,
        PromotionType type,
        LocalDateTime dateDebut,
        LocalDateTime dateFin,
        Produit produit
) {}
